pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/SumPrecisionAggregationFunction.java

```diff
@@ -48,8 +51,11 @@
 public class SumPrecisionAggregationFunction extends BaseSingleInputAggregationFunction<BigDecimal, BigDecimal> {
   private final Integer _precision;
   private final Integer _scale;
+  private final boolean _nullHandlingEnabled;
 
-  public SumPrecisionAggregationFunction(List<ExpressionContext> arguments) {
+  private final Set<Integer> _groupKeysWithNonNullValue;
```

Comment:
We don't need this set. This method should be similar to the `AVG`

Reply:
Just noticed that AVG also needs to return null when all input is null. Updated both `AVG` and `SumPrecision`:

```
SELECT avg(id) as avg, key FROM (VALUES (null, 1), (null, 1), (null, 2), (1, 3), (null, 3)) AS t(id, key)  GROUP BY key ORDER BY key DESC;
 avg  | key
------+-----
  1.0 |   3
 NULL |   2
 NULL |   1
```

Reply:
This one is still not fixed

Reply:
It is fixed in my local branch .. I usually mark comments as fixed as soon as I change locally, then send a new iteration.